spec/features/staff/users_can_create_a_budget_spec.rb

```diff
@@ -59,9 +56,7 @@
         fund_activity = create(:fund_activity, organisation: user.organisation)
         programme_activity = create(:programme_activity, parent: fund_activity, organisation: user.organisation)
 
-        visit activities_path
-
-        click_on(programme_activity.title)
+        visit organisation_activity_path(programme_activity.organisation, programme_activity)
 
         click_on(t("page_content.budgets.button.create"))
```

Comment:
I totally agree with your simplification of the test setup here: going straight to the start of the journey which you wish to exercise. 

**Given**: I am on the activity page
**When**: I fail to supply the required budget detail
**Then**: I see error messages describing what's missing

Where it aids readability, reuse, maintainability I would advocate using helper methods for any of these parts of the scenario...

Reply:
Are you proposing a helper method along the lines of `navigate_to_activity(activity)`?

Reply:
in retrospect I might be possible to have:

1. an initial commit  extracting all the preamble to get to the activity page to a method like that used in multiple places. No change to behaviour, just a refactoring of the test implementation

2. a following commit which then implements your change in routing behaviour in that new helper, **in one place** making the change much clearer

Reply:
I think I see what you are striking for and I agree, it would be much nicer.  Given I have the approval, are you happy to go ahead without attempting that - any other time I would have a go, I am struggling to find the energy on the project today though?